src/components/message-input/styles.scss

```diff
@@ -27,242 +27,182 @@
+    border: 1px solid theme.$color-greyscale-6;
+    border-radius: 8px;
+
+    &:focus-within {
+      border-color: theme.$color-primary-7;
     }
   }
 
-  &__giphy-icon {
+  &__text-and-emoji-wrapper {
     display: flex;
+    justify-content: flex-end;
     align-items: center;
-    justify-content: center;
-    height: 32px;
-    width: 32px;
+    gap: 8px;
+    flex-shrink: 0;
+    align-self: stretch;
   }
 
-  &__record-voice-icon {
+  &__mentions-text-area {
     display: flex;
-    align-items: center;
-    justify-content: center;
-    height: 32px;
-    width: 32px;
+    flex-direction: column;
+    flex: 1 0 0;
+    font-size: 14px;
+    line-height: 17px;
+
+    &__wrap {
+      &__suggestions {
+        top: unset !important;
+        bottom: 100%;
+      }
+    }
   }
 
-  &__emoji-picker {
-    position: absolute;
-    right: 0px;
-    bottom: 50px;
-    z-index: 1;
-  }
-}
+  &__mentions-text-area-wrap {
+    position: relative;
+    overflow-y: visible;
 
-.giphy__container {
-  width: 500px;
-  position: absolute;
-  bottom: 40px;
-  right: 0px;
-  background-color: themeDeprecated.$background-color-tertiary;
-  padding: 5px;
-  border-radius: 5px;
-  border: 1px solid themeDeprecated.$input-arrow-color;
+    &__highlighter {
+      position: relative;
+      box-sizing: border-box;
+      width: 100%;
+      color: transparent;
+      overflow: hidden;
+      white-space: pre-wrap;
+      background: transparent;
+      overflow-wrap: break-word;
+      border: 1px solid transparent;
+      text-align: start;
+
+      &__substring {
+        visibility: hidden;
+      }
 
-  &-grid {
-    height: 440px;
-    overflow-y: overlay;
-    margin-top: 5px;
-    .loader {
-      margin-top: 40px;
-      opacity: 1 !important;
+      @include tag-highlight;
     }
 
-    .giphy-gif {
-      cursor: pointer;
+    textarea {
+      color: theme.$color-greyscale-12;
+      word-break: break-word;
+      font-size: 14px !important;
+      line-height: 17px;
+      font-weight: 500;
+      outline: none;
+      border: none;
     }
-  }
 
-  &-search {
-    background-color: themeDeprecated.$input-background-primary-color !important;
-    border: 1px solid themeDeprecated.$input-arrow-color !important;
-    box-sizing: content-box;
-    padding-left: 5px;
-    border-radius: 5px !important;
-    height: 30px !important;
+    &__input {
+      display: block;
+      width: 100%;
+      position: absolute;
+      margin: 0px;
+      top: 0px;
+      left: 0px;
+      box-sizing: border-box;
+      background-color: transparent;
+      font-family: inherit;
+      font-size: inherit;
+      letter-spacing: inherit;
+      height: 100%;
+      bottom: 0px;
+      overflow: hidden;
+      resize: none;
+    }
 
-    > div {
-      height: 32px !important;
-      width: 32px !important;
-      border-radius: 5px !important;
-      margin: -1px -1px 0 0 !important;
+    &__suggestions {
+      background: none !important;
```

Comment:
I am a little confused where we should use `!important` & where we should not use it. Could you please explain that..? 🤔

Reply:
We use !important when we want to make sure that a certain declaration will always apply, no matter what other stylesheets or styles might also be applicable to that HTML element.

Here !important is being used to override the top and background properties of &__suggestions and the font-size of the textarea. These styles are getting applied from elsewhere in the CSS and we want to ensure these specific styles apply in these cases i.e. for suggestions class.

Use of !important should be limited though as it can make debugging difficult at times.

Reply:
ahhh, so you're basically overriding that style..got it. Thanks!